great_expectations/core/usage_statistics/usage_statistics.py

```diff
@@ -477,6 +483,47 @@ def get_batch_list_usage_statistics(data_context, *args, **kwargs):
     return payload
 
 
+# noinspection PyUnusedLocal
+def get_checkpoint_run_usage_statistics(checkpoint, *args, **kwargs):
+    try:
+        data_context_id = checkpoint.data_context.data_context_id
+    except AttributeError:
+        data_context_id = None
+    anonymizer = _anonymizers.get(data_context_id, None)
+    if anonymizer is None:
+        anonymizer = Anonymizer(data_context_id)
+        _anonymizers[data_context_id] = anonymizer
+    payload = {}
+
+    if checkpoint._usage_statistics_handler:
+        # noinspection PyBroadException
+        try:
+            checkpoint_run_anonymizer: CheckpointRunAnonymizer = (
+                checkpoint._usage_statistics_handler._checkpoint_run_anonymizer
+            )
+
+            checkpoint_config: CheckpointConfig = copy.deepcopy(checkpoint.config)
+
+            substituted_runtime_config: CheckpointConfig = (
+                checkpoint_run_anonymizer.resolve_config_using_acceptable_arguments(
+                    *(checkpoint,), **kwargs
+                )
+            )
+            resolved_runtime_kwargs: dict = substituted_runtime_config.to_json_dict()
+
+            payload = checkpoint_run_anonymizer.anonymize_checkpoint_run(
+                *(checkpoint,), **resolved_runtime_kwargs
+            )
+
+            checkpoint._config = checkpoint_config
```

Comment:
Is this to reset the checkpoint._config back to the original state after substitution / creation of payload on lines 507-516?

Reply:
@anthonyburdi Yes!  This is exactly what it is!